src/foam/lib/parse/Seq3.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.lib.parse;
+
+import foam.lib.json.KeyParser;
+
+public class Seq3 implements Parser {
+  private Parser[] parsers;
+  private int      index;
+  private int      opIndex;
+  private Object   operator;
+
+  public Seq3(int i, int j, Parser... args) {
```

Comment:
This class is hard coded to pass something called "operator" through.  I would expect Seq3 to take 3 index values, just like Seq2 takes 2 values, and Seq1 takes 1 value, and Seq0 takes 0 values.

If we take int i, int j and int k, and pass all three values via ps.setValue(), then we don't need to modify PStream to support operator()/setOperator()